monai/data/image_reader.py

```diff
@@ -235,11 +236,28 @@ def _get_array_data(self, img) -> np.ndarray:
         """
         Get the raw array data of the image, converted to Numpy array.
 
+        Following PyTorch conventions, the returned array data has contiguous channels,
+        e.g. for an RGB image, all red channel image pixels are contiguous in memory.
+        The first axis of the returned array is the channel axis.
+
         Args:
             img: a ITK image object loaded from a image file.
 
         """
-        return itk.array_view_from_image(img, keep_axes=False)
+        channels = img.GetNumberOfComponentsPerPixel()
+        if channels == 1:
+            return itk.array_view_from_image(img, keep_axes=False)
+        else:
+            # The memory layout of itk.Image has all pixel's channels adjacent
```

Comment:
Hi @ericspod and @thewtex ,

I am not very sure about the dim order in medical images, I think we discussed this topic before for ITK and Nibabel things. Could you please help share some comments here?

Thanks.

Reply:
There is not a consistent dim order for all medical images.

Reply:
That's true, it's not consistent. With Nifti files it's almost always been in my experience WHDT or HWDT, but nothing enforces this.

Reply:
Thanks @hjmjohnson and @ericspod for your comments, so what do you think about this PR?

Reply:
I think this is correct for colour images like the test demonstrates, the issue of ordering of spatial and temporal dimensions might have to be up to the user.